graphistry/compute/hop.py

```diff
@@ -141,7 +141,11 @@ def hop(self: Plottable,
     matches_edges = edges_indexed[[EDGE_ID]][:0]
 
     #richly-attributed subset for dest matching & return-enriching
-    base_target_nodes = target_wave_front if target_wave_front is not None else g2._nodes
+    if target_wave_front is None:
+        base_target_nodes = g2._nodes
+    else:
+        base_target_nodes = concat([target_wave_front, g2._nodes], ignore_index=True, sort=False).drop_duplicates(subset=[g2._node])
```

Comment:
this confuses me a bit, what is `g2._nodes`? is that all nodes in the graph? or is it something else?

if it is all nodes than this line would be equivalent of `base_target_nodes = g2._nodes` i think...

Reply:
`g2._nodes` is the materialized `self`._nodes`:

* on forwards-pass of chain, that's all graph nodes, even if user didn't materialize first
* on reverse-pass, nodes are already materialized, so this is the nodes that were reached on the forwards pass
  *  it does NOT include the starting nodes from the forwards pass, so we include those later